docs/source/topics/backends/snmp.rst

```diff
@@ -0,0 +1,149 @@
+the tree gains more depth.
+Obviously, by getting the smallest MIB which is "1" or simply " . " one can get all the tree.
+
+The base SNMP backend contains the following methods (some internal methods are not documented and are subject to change in the future):
+
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **to_dict**  | Returns a dict containing monitoring information depending on the type of the device.                                                    |
+|              | It follows the `NetJSON Devicemonitoring <https://netjson.org/rfc.html#name-devicemonitoring>`_ spec                                     |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **to_json**  | Calls the `to_dict` method and returns a JSON string of the dict                                                                         |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **validate** | Checks if connection with the device is working and raises `NetengineError` in case something is wrong                                   |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+
+Initializing an SNMP backend class requires the following arguments:
+
++---------------+---------------------------------------------------------------------+
+| **host**      | Management ip or hostname of the device                             |
++---------------+---------------------------------------------------------------------+
+| **community** | Community string for the SNMP connection. Default value is 'public' |
++---------------+---------------------------------------------------------------------+
+| **agent**     | Agent string for the SNMP connection                                |
++---------------+---------------------------------------------------------------------+
+| **port**      | Port for the SNMP connection. Default value is `161`                |
++---------------+---------------------------------------------------------------------+
+
+The SNMP backend provides support for 2 firmwares:
+ * AirOS
+ * OpenWRT
+
+.. note::
+
+    The data collected by Netengine is dependant on the OIDs available on your device. Some proprietary manufacturers may not
+    provide the same information as others.
+
+*****
+AirOS
+*****
+
+With AirOS, Netengine is able to collect the following information which is returned in the
+`NetJSON Devicemonitoring <https://netjson.org/rfc.html#name-devicemonitoring>`_ format:
+
++------------+------------------------------------------------------------------------------------------------------------+
```

Comment:
please let's switch from table sytnax which is painful to maintain to simple headings and list after headings.